test/TokenPair.test.ts

```diff
@@ -60,378 +56,355 @@ export async function mint(
+      liquidity: liquidity
+    })
+
+    const tokenPairState = getContractState<TokenPairTypes.Fields>(burnResult.contracts, fixture.contractId)
+    expect(tokenPairState.fields.totalSupply).toEqual(MinimumLiquidity)
+    expect(tokenPairState.fields.reserve0).toEqual(MinimumLiquidity)
+    expect(tokenPairState.fields.reserve1).toEqual(MinimumLiquidity)
+    expect(contractBalanceOf(tokenPairState, token0Id)).toEqual(MinimumLiquidity)
+    expect(contractBalanceOf(tokenPairState, token1Id)).toEqual(MinimumLiquidity)
+
+    const senderOutputs = burnResult.txOutputs.filter((o) => o.address === sender && o.tokens !== undefined && o.tokens.length > 0)
+    expect(senderOutputs.length).toEqual(2)
+    const senderTokens = [...senderOutputs[0].tokens!, ...senderOutputs[1].tokens!]
+    expectTokensEqual(senderTokens, [
+      { id: token0Id, amount: token0Amount - MinimumLiquidity },
+      { id: token1Id, amount: token1Amount - MinimumLiquidity },
+    ])
+  })
 
-    async function test(token0Id: string, token1Id: string) {
-      const fixture = createTokenPair(token0Id, token1Id)
-      const { contractState, reserve0, reserve1 } = await mint(fixture, sender, 100n, 80000n)
-      await expectAssertionError(
-        testSwap(fixture, token0Id, reserve0, reserve1, contractState.fields, contractState.asset),
-        fixture.address,
-        ErrorCodes.InsufficientLiquidity
-      )
-      await expectAssertionError(
-        testSwap(fixture, token1Id, reserve1, reserve0, contractState.fields, contractState.asset),
-        fixture.address,
-        ErrorCodes.InsufficientLiquidity
-      )
-      await expectAssertionError(
-        testSwap(fixture, token0Id, 10n, 0n, contractState.fields, contractState.asset),
-        fixture.address,
-        ErrorCodes.InsufficientOutputAmount
-      )
-      await expectAssertionError(
-        testSwap(fixture, randomTokenId(), 10n, 10n, contractState.fields, contractState.asset),
-        fixture.address,
-        ErrorCodes.InvalidTokenInId
-      )
-
-      const amountIn0 = 20n // token0Id
-      const expectedAmountOut0 = getAmountOut(amountIn0, reserve0, reserve1)
-      await testSwap(fixture, token0Id, amountIn0, expectedAmountOut0, contractState.fields, contractState.asset)
-      await expectAssertionError(
-        testSwap(fixture, token0Id, amountIn0, expectedAmountOut0 + 1n, contractState.fields, contractState.asset),
-        fixture.address,
-        ErrorCodes.InvalidK
-      )
-
-      const amountIn1 = 60000n // token1Id
-      const expectedAmountOut1 = getAmountOut(amountIn1, reserve1, reserve0)
-      await testSwap(fixture, token1Id, amountIn1, expectedAmountOut1, contractState.fields, contractState.asset)
-      await expectAssertionError(
-        testSwap(fixture, token1Id, amountIn1, expectedAmountOut1 + 1n, contractState.fields, contractState.asset),
-        fixture.address,
-        ErrorCodes.InvalidK
-      )
+  it('price{0,1}CumulativeLast', async () => {
```

Comment:
This test is not very good, maybe we should add an optional parameter `blockTimestamp` to the test contract endpoint, what do you think?

Reply:
Yessss, please add it for us 🙏